strawberry/schema/schema_converter.py

```diff
@@ -0,0 +1,324 @@
+from typing import Any, Callable, Dict, Type, Union, cast
+
+from graphql import (
+    GraphQLArgument,
+    GraphQLDirective,
+    GraphQLEnumType,
+    GraphQLEnumValue,
+    GraphQLField,
+    GraphQLInputField,
+    GraphQLInputObjectType,
+    GraphQLInputType,
+    GraphQLInterfaceType,
+    GraphQLList,
+    GraphQLNonNull,
+    GraphQLNullableType,
+    GraphQLObjectType,
+    GraphQLOutputType,
+    GraphQLScalarType,
+    GraphQLType,
+    GraphQLUnionType,
+    Undefined,
+)
+
+from strawberry.arguments import UNSET
+from strawberry.directive import DirectiveDefinition
+from strawberry.enum import EnumDefinition, EnumValue
+from strawberry.field import FieldDefinition
+from strawberry.resolvers import get_resolver
+from strawberry.scalars import is_scalar
+from strawberry.types.types import ArgumentDefinition, TypeDefinition, undefined
+from strawberry.union import StrawberryUnion
+
+from .types.concrete_type import ConcreteType
+from .types.scalar import get_scalar_type
+
+
+FIELD_ARGUMENT_TYPE = Union[FieldDefinition, ArgumentDefinition]
+
+
+class GraphQLCoreConverter:
+    # TODO: Make abstract
+
+    def __init__(self):
+        self.type_map: Dict[str, ConcreteType] = {}
+
+    def get_graphql_type_field(self, field: FIELD_ARGUMENT_TYPE) -> GraphQLType:
+        # TODO: Completely replace with get_graphql_type
+
+        graphql_type: GraphQLType
+
+        if _is_list(field):
+            graphql_type = self.from_list(field)
```

Comment:
I like this `from_abc` method 👏

Reply:
Hopefully they can all be made abstract, such that any backend we support could implement each of them